cg/meta/transfer/external_data.py

```diff
@@ -153,9 +153,9 @@ def curate_sample_folder(self, cust_name: str, force: bool, sample_folder: Path)
         """Changes the name of the folder to the internal_id. If force is true replaces any previous folder."""
         customer: Customer = self.status_db.get_customer_by_customer_id(customer_id=cust_name)
         customer_folder: Path = sample_folder.parent
-        sample: Sample = self.status_db.find_samples(
+        sample: Sample = self.status_db.get_samples_by_customer_and_name(
             customer=customer, name=sample_folder.name
-        ).first()
+        )[0]
```

Comment:
What if no sample is found? We might want to add some error handling, the code before works for that case but would cause an index error here.

Reply:
I think this was a logic error on my side. thef function should just return 1 single sample not a list of sample